index.ts

```diff
@@ -1,32 +1,39 @@
-import { device, DeviceOptions } from 'aws-iot-device-sdk';
-import { Subject, Observer, Observable } from 'rxjs';
-import * as AWS from 'aws-sdk';
-import { v4 as uuidV4 } from 'uuid';
-import * as pako from 'pako';
+import { device, DeviceOptions } from "aws-iot-device-sdk";
+import { Subject, Observer, Observable } from "rxjs";
+import * as AWS from "aws-sdk";
+import { v4 as uuidV4 } from "uuid";
+import * as pako from "pako";
 
 export default class AwsIot {
-  public readonly events = new Subject<IotEvent>();
+  get events(): Observable<IotEvent> {
+    return this._events.asObservable();
+  }
+
+  get topicsForTest(): Array<IDeferredTopic> {
+    return this._deferredTopics;
+  } 
 
-  private client!: device;
-  private topics = new Array<string>();
+  private _client!: device;
```

Comment:
What is that exclamation mark for? 

```suggestion
  private _client: device;
```

Reply:
Is is a "can not be null" mark?

Reply:
This is not related to my changes and I don't remember why I had to put that there but the exclamation mark is a [non-null-assertion-operator](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator).

Could most likely be removed since I think it had something to do with babel which is removed with these changes but I'm not certain.

Reply:
Checked and it can't be removed since the TS compiler will complain that the property is not instantiated in the constructor so leaving this in.